src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -353,6 +351,15 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         }
     }
 
+    // E0120: Drop can be only implemented by structs and enums
+    private fun checkImplDropForNonAdtError(holder: AnnotationHolder, impl: RsImplItem, traitRef: RsTraitRef, trait: RsTraitItem) {
+        if (trait != trait.knownItems.Drop) return
+
+        if (impl.typeReference?.type is TyAdt) return
```

Comment:
Looks dangerous for the case of Drop impl with missed type reference, i.e.
```rust
impl Drop for {}
```

Let's add a test that show that we don't have an error in this case.